### PR TITLE
Updated circle ci to execute spotless

### PR DESCRIPTION
**Pull request by rain-on**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -38,7 +38,11 @@ jobs:
           name: Downloading dependencies
           command: ./gradlew dependencies checkLicenses
 
-      - run:
+      - compile :
```
Comment (CjHare) [inline .circleci/config.yml]: Doesn't `build` also execute `assemble`? _i.e. no need for both?_

Comment (jframe) [reply .circleci/config.yml]: Are there other checks we don't want to be run other than test? Maybe ./gradlew assemble spotlessCheck is more what we are after here.

Comment (ajsutton) [reply .circleci/config.yml]: Its usually good if CI runs a single target (typically `./gradlew build` though sometimes the default is a subset of what CI does and CI might run `./gradlew full-build` or something) and then that target is configured as an alias in the Gradle scripts to include whatever targets you want.  That way it's easy for anyone to run the exact same checks as CI.

Sometime parallelisation prevents doing this but its still good to keep the CI config as simple as possible and put as much of the definition into the gradle files developers use.

Comment (rain-on) [reply .circleci/config.yml]: That sounds awesome - for now we really need spotless checks in, and this gives us an "early" exit on fail which is sort of nice...

**Pull request merged**